orderbook/src/account_balances.rs

```diff
@@ -605,35 +465,31 @@ mod tests {
             Some(vault.clone()),
             allowance_target.address(),
             H160::from_low_u64_be(1),
-            NoopMetrics::arc(),
-        );
-
-        // Not available until registered
-        assert_eq!(
-            fetcher.get_balance(trader.address(), token.address(), SellTokenSource::External),
-            None,
         );
 
-        fetcher
-            .register(trader.address(), token.address(), SellTokenSource::External)
-            .await;
+        let get_balance = || async {
+            fetcher
+                .get_balances(&[Query {
+                    owner: trader.address(),
+                    token: token.address(),
+                    source: SellTokenSource::External,
+                }])
+                .await
+                .into_iter()
+                .next()
+                .unwrap()
+                .unwrap()
+        };
 
-        assert_eq!(
-            fetcher.get_balance(trader.address(), token.address(), SellTokenSource::External),
-            Some(U256::zero()),
-        );
+        assert_eq!(get_balance().await, U256::zero());
 
         // Balance without allowance and approval should not affect available balance
         token
             .mint(trader.address(), 100.into())
             .send()
             .await
             .unwrap();
-        fetcher.update().await;
-        assert_eq!(
-            fetcher.get_balance(trader.address(), token.address(), SellTokenSource::External),
-            Some(U256::zero()),
-        );
+        assert_eq!(get_balance().await, U256::zero(),);
```

Comment:
ubernit: Here and in a few more places:

```suggestion
        assert_eq!(get_balance().await, U256::zero());
```